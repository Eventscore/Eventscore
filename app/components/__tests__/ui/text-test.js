/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import { Alert } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { Text } from '@ui';

it('Text renders correctly', () => {
  const tree = renderer.create(
    <Text>Hey Now...</Text>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Text P renders correctly', () => {
  const tree = renderer.create(
    <Text p>Hey Now...</Text>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Text H1 renders correctly', () => {
  const tree = renderer.create(
    <Text h1>Hey Now...</Text>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Text H2 renders correctly', () => {
  const tree = renderer.create(
    <Text h2>Hey Now...</Text>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Text H3 renders correctly', () => {
  const tree = renderer.create(
    <Text h3>Hey Now...</Text>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Text H4 renders correctly', () => {
  const tree = renderer.create(
    <Text h4>Hey Now...</Text>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Text H5 renders correctly', () => {
  const tree = renderer.create(
    <Text h5>Hey Now...</Text>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Text H1 with Link renders correctly', () => {
  const tree = renderer.create(
    <Text h1 onPress={() => Alert.alert('Testing')}>Hey Now...</Text>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
