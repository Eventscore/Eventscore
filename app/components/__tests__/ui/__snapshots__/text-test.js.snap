// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text H1 renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  h1={true}
  h2={false}
  h3={false}
  h4={false}
  h5={false}
  onPress={null}
  p={false}
  style={
    Array [
      Object {
        "color": "#0E4EF8",
        "fontFamily": "HelveticaNeue",
        "fontSize": 24.5,
        "fontWeight": "800",
        "left": 0,
        "lineHeight": 30,
        "margin": 0,
        "marginBottom": 4,
        "right": 0,
      },
    ]
  }
>
  Hey Now...
</Text>
`;

exports[`Text H1 with Link renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  h1={true}
  h2={false}
  h3={false}
  h4={false}
  h5={false}
  onPress={[Function]}
  p={false}
  style={
    Array [
      Object {
        "color": "#0E4EF8",
        "fontFamily": "HelveticaNeue",
        "fontSize": 24.5,
        "fontWeight": "800",
        "left": 0,
        "lineHeight": 30,
        "margin": 0,
        "marginBottom": 4,
        "right": 0,
      },
      Object {
        "color": "#0E4EF8",
        "textDecorationLine": "underline",
      },
    ]
  }
>
  Hey Now...
</Text>
`;

exports[`Text H2 renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  h1={false}
  h2={true}
  h3={false}
  h4={false}
  h5={false}
  onPress={null}
  p={false}
  style={
    Array [
      Object {
        "color": "#0E4EF8",
        "fontFamily": "HelveticaNeue",
        "fontSize": 21,
        "fontWeight": "800",
        "left": 0,
        "lineHeight": 26,
        "margin": 0,
        "marginBottom": 4,
        "right": 0,
      },
    ]
  }
>
  Hey Now...
</Text>
`;

exports[`Text H3 renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  h1={false}
  h2={false}
  h3={true}
  h4={false}
  h5={false}
  onPress={null}
  p={false}
  style={
    Array [
      Object {
        "color": "#0E4EF8",
        "fontFamily": "HelveticaNeue",
        "fontSize": 17.5,
        "fontWeight": "500",
        "left": 0,
        "lineHeight": 23,
        "margin": 0,
        "marginBottom": 4,
        "right": 0,
      },
    ]
  }
>
  Hey Now...
</Text>
`;

exports[`Text H4 renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  h1={false}
  h2={false}
  h3={false}
  h4={true}
  h5={false}
  onPress={null}
  p={false}
  style={
    Array [
      Object {
        "color": "#0E4EF8",
        "fontFamily": "HelveticaNeue",
        "fontSize": 15.400000000000002,
        "fontWeight": "800",
        "left": 0,
        "lineHeight": 23,
        "margin": 0,
        "marginBottom": 4,
        "right": 0,
      },
    ]
  }
>
  Hey Now...
</Text>
`;

exports[`Text H5 renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  h1={false}
  h2={false}
  h3={false}
  h4={false}
  h5={true}
  onPress={null}
  p={false}
  style={
    Array [
      Object {
        "color": "#0E4EF8",
        "fontFamily": "HelveticaNeue",
        "fontSize": 14,
        "fontWeight": "800",
        "left": 0,
        "lineHeight": 18,
        "margin": 0,
        "marginBottom": 4,
        "marginTop": 4,
        "right": 0,
      },
    ]
  }
>
  Hey Now...
</Text>
`;

exports[`Text P renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  h1={false}
  h2={false}
  h3={false}
  h4={false}
  h5={false}
  onPress={null}
  p={true}
  style={
    Array [
      Object {
        "color": "#222222",
        "fontFamily": "HelveticaNeue",
        "fontSize": 14,
        "fontWeight": "500",
        "lineHeight": 18,
        "marginBottom": 8,
      },
    ]
  }
>
  Hey Now...
</Text>
`;

exports[`Text renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  h1={false}
  h2={false}
  h3={false}
  h4={false}
  h5={false}
  onPress={null}
  p={false}
  style={
    Array [
      Object {
        "color": "#222222",
        "fontFamily": "HelveticaNeue",
        "fontSize": 14,
        "fontWeight": "500",
        "lineHeight": 18,
      },
    ]
  }
>
  Hey Now...
</Text>
`;
